pkg/cloudprovider/providers/gce/gce_annotations.go

```diff
@@ -57,61 +61,144 @@ const (
 	NetworkTierDefault  NetworkTier = NetworkTierPremium
 )
 
-// GetLoadBalancerAnnotationType returns the type of GCP load balancer which should be assembled.
-func GetLoadBalancerAnnotationType(service *v1.Service) (LoadBalancerType, bool) {
-	v := LoadBalancerType("")
-	if service.Spec.Type != v1.ServiceTypeLoadBalancer {
-		return v, false
+type lbAnnotation struct {
+	// values includes all valid values for the annotation key.
+	values sets.String
+	// defaultValue is the default value for the annotation key.
+	defaultValue string
+}
+
+var (
+	// allServiceLBAnnotations defines all supported load balancer annoations
+	// for Service.
+	allServiceLBAnnotations = map[string]*lbAnnotation{
+		ServiceAnnotationLoadBalancerType: {
+			values: sets.NewString(
+				ServiceAnnotationLoadBalancerTypeInternal,
+				ServiceAnnotationLoadBalancerTypeExternal,
+				"internal", // The lower-case value has been deprecated.
+			),
+			defaultValue: ServiceAnnotationLoadBalancerTypeExternal,
+		},
+		ServiceAnnotationILBBackendShare: {
+			values:       sets.NewString(AnnotationTrueValue, AnnotationFalseValue),
+			defaultValue: AnnotationFalseValue,
+		},
+		NetworkTierAnnotationKey: {
+			values:       sets.NewString(NetworkTierAnnotationPremium, NetworkTierAnnotationStandard),
+			defaultValue: NetworkTierAnnotationPremium,
+		},
 	}
 
-	l, ok := service.Annotations[ServiceAnnotationLoadBalancerType]
-	v = LoadBalancerType(l)
-	if !ok {
-		return v, false
+	// deprecatedServiceAnnotations maps the deprecated key to its replacement.
+	// The deprecation keys are not defined as constants on purpose to avoid
+	// accidental uses.
+	deprecatedServiceLBAnnotations = map[string]string{
+		// This key did not correctly specify "alpha" and has been deprecated.
+		"cloud.google.com/load-balancer-backend-share": ServiceAnnotationLoadBalancerType,
 	}
 
-	switch v {
-	case LBTypeInternal, deprecatedTypeInternalLowerCase:
-		return LBTypeInternal, true
-	default:
-		return v, false
+	// ILBAnnotationKeys is a set of all supported annotation keys for an
+	// internal load balancer.
+	ILBAnnotationKeys = sets.NewString(ServiceAnnotationILBBackendShare)
```

Comment:
Do we have to expose `ILBAnnotationKeys` and `ELBAnnotationKeys`?

Reply:
And may be in future we will have `commonAnnotationKeys` that indicates annotation keys that are supported on both ELB and ILB so we don't need to duplicate them in two places. Just a thought :)

Reply:
> Do we have to expose ILBAnnotationKeys and ELBAnnotationKeys?

Nope. I do think `providers/gce` can use more subpackages. If we ever do that, we might want to export these.

Reply:
> And may be in future we will have commonAnnotationKeys that indicates annotation keys that are supported on both ELB and ILB so we don't need to duplicate them in two places. Just a thought :)

Depends. I think it's appealing to keep individual sets so that it's easy to manipulate the sets if needed. For example, what if we add a new LB type and we'll need more complex validation logic? Of course, this is mot very likely so maybe keeping a common set makes more sense.